flink-table/flink-table-planner/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -2024,7 +2029,9 @@ private RexNode simplifySearch(RexCall call, RexUnknownAs unknownAs) {
     private RexNode simplifyCast(RexCall e) {
         RexNode operand = e.getOperands().get(0);
         operand = simplify(operand, UNKNOWN);
-        if (sameTypeOrNarrowsNullability(e.getType(), operand.getType())) {
+        if (sameTypeOrNarrowsNullability(e.getType(), operand.getType())
+                || e.getType() instanceof RawRelDataType) {
+            // || e.getType().getSqlTypeName() == SqlTypeName.OTHER) {
```

Comment:
Maybe we need this instead?

Reply:
Not sure, OTHER catches other types in our type system, right?

Reply:
For us, currently, only `RAW` is associated with `OTHER`